articles/azure-resource-manager/managed-applications/create-uidefinition-functions.md

```diff
@@ -481,6 +481,48 @@ Assume `element1` and `element2` are undefined. The following example returns `"
 "[coalesce(steps('foo').element1, steps('foo').element2, 'foobar')]"
 ```
 
+   > [!NOTE]
+   > This function is especially in the context of optional invocation that happens due to user action after the page loads.  For example, if the constraints placed on one field in the UI depend on the currently selected value of another, **initially non-visible** field.  In this case, `coalesce()` can be used to allow the function to be syntactically valid at page load time, while having the desired effect when the user interacts with the field.
+   >
+   > Consider this `DropDown` which allows the user to choose from several different database types.
+   >
+   > ```
+   > {
+   >     "name": "databaseType",
+   >     "type": "Microsoft.Common.DropDown",
+   >     "label": "Choose database type",
+   >     "toolTip": "Choose database type",
+   >     "defaultValue": "Oracle Database",
+   >     "visible": "[bool(steps('section_database').connectToDatabase)]"
+   >     "constraints": {
+   >         "allowedValues": [
+   >             {
+   >                 "label": "Azure Database for PostgreSQL",
+   >                 "value": "postgresql"
+   >             },
+   >             {
+   >                 "label": "Oracle Database",
+   >                 "value": "oracle"
+   >             },
+   >             {
+   >                 "label": "Azure SQL",
+   >                 "value": "sqlserver"
+   >             }
+   >         ],
+   >         "required": true
+   >     },
+   > ```
+   >
+   > If we want to condition the action of another field on the current
```

Comment:
General guidance is to not use "we" so this part could revised: "To condition the action of another field on the current value of this field, use `coalesce()` ..."

Reply:
I agree with your modification.